internal/services/managedhsm/resourceids.go

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package managedhsm
+
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=ManagedHSMNestedKeyWithVersion -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/managedHSMs/mhsm1/keys/key1/versions/version1
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=ManagedHSMNestedKeyVersionless -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/managedHSMs/mhsm1/keys/key1
```

Comment:
The legacy Resource ID generator shouldn't be used for new Resources going forwards - can we instead update these to use [the new Resource ID interface](https://github.com/hashicorp/go-azure-helpers/blob/891641b735c70484dffd36a0fba21bca79efc603/resourcemanager/resourceids/interface.go#L6-L20) that `hashicorp/go-azure-sdk` etc uses?

Reply:
Also shouldn't these be the Data Plane IDs rather than the Resource Manager IDs?